core/src/main/java/org/mule/util/xmlsecurity/DefaultXMLSecureFactories.java

```diff
@@ -69,4 +74,66 @@ public static XMLInputFactory createXmlInputFactory(Boolean externalEntities, Bo
 
         return factory;
     }
+
+    public static TransformerFactory createTransformerFactory(Boolean externalEntities, Boolean expandEntities)
+    {
+        TransformerFactory factory = TransformerFactory.newInstance();
+
+        configureTransformerFactory(externalEntities, expandEntities, factory);
+
+        return factory;
+    }
+
+    public static void configureTransformerFactory(Boolean externalEntities, Boolean expandEntities, TransformerFactory factory)
+    {
+        if (!externalEntities && !expandEntities)
+        {
+            try
+            {
+                factory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, "");
+                factory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+            }
+            catch (Exception e)
+            {
+                logWarning("TransformerFactory", factory.getClass().getName());
+            }
+        }
+    }
+
+    public static void configureSchemaFactory(Boolean externalEntities, Boolean expandEntities, SchemaFactory factory)
```

Comment:
We don't want them to configure manually others because it would cause performance issues (configuration was the expensive operation, more than creating factories).